tiny_dnn/core/kernels/fully_connected_op.h

```diff
@@ -24,30 +24,38 @@ class FullyConnectedOp : public core::OpKernel {
     auto params = OpKernel::params_->fully();
 
     // incomimg/outcoming data
-    const tensor_t &in_data = context.input(0);
-    const tensor_t &W       = context.input(1);
-    const tensor_t *bias    = params.has_bias_ ? &context.input(2) : nullptr;
-    tensor_t &out_data      = context.output(0);
+    // TODO(Randl): Remove once layers forward and backward by themself.
+    const Tensor<float_t> in_data(context.input(0)), weights(context.input(1)),
+      bias = params.has_bias_ ? Tensor<float_t>(context.input(2))
+                              : Tensor<float_t>();
+    Tensor<float_t> out_data(context.output(0));
 
     // initialize outputs
-    fill_tensor(out_data, float_t{0});
+    out_data.fill(0);
 
     // call the algorithm depending  on the selected engine type
 
     const core::backend_t engine = context.engine();
 
     if (engine == core::backend_t::internal) {
-      kernels::fully_connected_op_internal(
-        in_data, W[0], params.has_bias_ ? (*bias)[0] : vec_t(), out_data,
-        params, context.parallelize());
+      kernels::fully_connected_op_internal(in_data, weights, bias, out_data,
+                                           params.has_bias_,
+                                           context.parallelize());
+
+      // TODO(Randl): Remove once layers forward and backward by themself.
+      context.output(0) = out_data.toTensor();
     } else if (engine == core::backend_t::nnpack) {
-      kernels::fully_connected_op_nnpack(
-        in_data, W[0], params.has_bias_ ? (*bias)[0] : vec_t(), out_data,
-        params, context.parallelize());
+      kernels::fully_connected_op_avx(in_data, weights, bias, out_data,
```

Comment:
In here, this should be `fully_connected_op_nnpack`, not `fully_connected_op_avx`.